src/main/java/uk/gov/justice/hmpps/prison/service/OffenderActivitiesService.java

```diff
@@ -0,0 +1,53 @@
+package uk.gov.justice.hmpps.prison.service;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.justice.hmpps.prison.api.model.OffenderActivities;
+import uk.gov.justice.hmpps.prison.api.model.OffenderActivitySummary;
+import uk.gov.justice.hmpps.prison.repository.jpa.model.OffenderProgramProfile;
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.OffenderProgramProfileRepository;
+
+import java.time.LocalDate;
+import java.util.stream.Collectors;
+
+@Service
+@Transactional(readOnly = true)
+public class OffenderActivitiesService {
+    private final OffenderProgramProfileRepository offenderProgramProfileRepository;
+    private final BookingService bookingService;
+
+    public OffenderActivitiesService(final OffenderProgramProfileRepository offenderProgramProfileRepository, final BookingService bookingService) {
+        this.offenderProgramProfileRepository = offenderProgramProfileRepository;
+        this.bookingService = bookingService;
+    }
+
+    public OffenderActivities getCurrentWorkActivities(String offenderNo) {
+        final var currentBookingId = bookingService.getLatestBookingByOffenderNo(offenderNo).getBookingId();
+        final var allAllocatedWorkActivities = offenderProgramProfileRepository.findByOffenderBooking_BookingIdAndProgramStatus(currentBookingId, "ALLOC");
+        final var workActivities =  allAllocatedWorkActivities.stream().filter(this::isValidCurrentActivity).map(this::transform).collect(Collectors.toList());
+        return OffenderActivities.builder()
+            .offenderNo(offenderNo)
+            .bookingId(currentBookingId)
+            .workActivities(workActivities)
+            .build();
+    }
+
+    private boolean isValidCurrentActivity(OffenderProgramProfile offenderProgramProfile) {
```

Comment:
I have put these checks here rather than in the JPA query for several reasons:

 - To prevent the repo from having lots of JPQL
 - I think we will need to reuse this logic for other queries
 - Easier to test (no SQL to set up the tables)

It might be that we have to move these checks to the SQL for performance, but for the moment they are better here.

Reply:
Maybe consider moving the method on to OffenderProgramProfile, it *might* be easier to reuse. Seems to be a pattern we are using more, get the JPA model to understand itself